VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsThreadFragment.java

```diff
@@ -200,7 +200,7 @@ public void onEventMainThread(DiscussionCommentPostedEvent event) {
 
     public void onEventMainThread(DiscussionThreadPostedEvent event) {
         // If a new post is created in this topic, insert it at the top of the list, after any pinned posts
-        if (!isFollowingTopics() && event.getDiscussionThread().getTopicId().equalsIgnoreCase(discussionTopic.getIdentifier())) {
+        if (!isFollowingTopics() && !isAllTopics() && event.getDiscussionThread().getTopicId().equalsIgnoreCase(discussionTopic.getIdentifier())) {
```

Comment:
Yup. I moved this to a new method; DiscussionTopic.containsThread which also properly accounts for nested topics.